src/js/components/DateInput/DateInput.js

```diff
@@ -74,8 +74,8 @@ const DateInput = forwardRef(
     useEffect(() => {
       if (
         schema &&
-        value &&
-        ((Array.isArray(value) && value[0]) || !Array.isArray(value))
+        ![undefined, null].includes(value) &&
+          (Array.isArray(value) || !Array.isArray(value))
```

Comment:
Won't this logic always return true, since the value is either an Array or not an Array?

I'm thinking `undefined !== value` might be enough. That would align with how `undefined` is what determines whether an input component is controlled or not. In other words, I don't think we need to also allow for `null`.

Reply:
You are absolutely right. 😅
I have modified it.